_drafts/code-properties-and-language-features.md

```diff
@@ -0,0 +1,50 @@
+I feel like I only just started learning good object-oriented design, and the world has moved on to functional programming! As a result, the OO principles that are fresh on my mind are exactly what people are moving away from. For example, as I've been checking out Elixir and Redux, I've had questions about whether a lack of certain OO features causes problems. For example, encapsulation allows you to isolate implementation knowledge to one part of your app, so that other parts of your app can use that functionality without being coupled to its implementation. Is it a problem that FP doesn't offer the same kind of encapsulation?
+
+It's difficult for me to find answers to these questions. As Yehuda Katz [tweeted][yehuda-1] [recently][yehuda-2], there's a tendency to describe programming advancements as radically new paradigms, and past approaches as terrible. There's no grappling with the arguments older approaches made.
+
+This leaves me with a conundrum: I bought into arguments about encapsulation increasing maintainability. Were those arguments true, or were they just something we made up to tell ourselves so we could build software in a way we liked?
+
+Even though these things aren't discussed much publicly, asking questions of people who know Elixir and Redux helped me with this.
+
+# Questioning Elixir and Redux
+
+First, I asked about the way data is separate from functionality in Elixir, and whether that caused problems. But I learned that when you define a struct inside a module, you also usually define functions that operate on that struct inside the same module. Because the name of the module and the name of the struct are the same, they go together naturally. So data and functionality aren't entirely separated.
```

Comment:
> Because the name of the module and the name of the struct are the same, they go together naturally. So data and functionality aren't entirely separated.

It's true that in Elixir, `String.upcase/1` is tied to strings; it wouldn't make sense to pass anything else. But `Enum.reverse` applies to any enumerable, and like in Ruby, it's preferable to put things there instead of in an Array-specific location if it might be possible to use them on other kinds of data.